src/test/compile-fail/union-field-privacy.rs

```diff
@@ -0,0 +1,21 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
```

Comment:
well, we can wait on the `ui` stuff -- we can do it all at once, probably. But let's definitely make a directory for `union` tests. (Though in some cases, e.g., with borrowck, and maybe things like privacy, I'd prefer to group `union` in with that.)